crates/cli-support/src/js/mod.rs

```diff
@@ -1168,10 +1169,37 @@ impl<'a> Context<'a> {
         } else {
             ""
         };
+
+        // Add a binding for using `TextEncoder#encodeInto` which avoids
+        // intermediate buffer allocations by encoding directly into wasm's
+        // memory. This is a very new API, however, so we feature detect it and
+        // fall back to the widely available `TextEncoder#encode` method.
+        self.global("
+            function passStringToWasmFast(string) {
+                let size = string.length;
+                let ptr = wasm.__wbindgen_malloc(size);
+                let writeOffset = 0;
+                while (true) {
+                    const view = getUint8Memory().subarray(ptr + writeOffset, ptr + size);
+                    const { read, written } = cachedTextEncoder.encodeInto(string, view);
+                    string = string.substring(read);
+                    writeOffset += written;
+                    if (string.length === 0)
+                        break;
+                    ptr = wasm.__wbindgen_realloc(ptr, size, size * 2);
+                    size *= 2;
+                }
+                WASM_VECTOR_LEN = writeOffset;
+                return ptr;
+            }
+        ");
+
         self.global(&format!(
             "
             function passStringToWasm(arg) {{
                 {}
+                if (typeof cachedTextEncoder.encodeInto === 'function')
+                    return passStringToWasmFast(arg);
```

Comment:
Instead of checking every time we pass a string, we should do this:

```js
(function () {
    return typeof TextEncoder.prototype.encodeInto === "function"
        ? passStringToWasmFast
        : passStringToWasm;

    function passStringToWasmFast(arg) { ... }
    function passStringToWasm(arg) { ... }
}())

Reply:
The `use_encode` body is still checking for `encodeInto`, which I think is an oversight

Reply:
Oops yes